File: cpp/Tree/102.binary-tree-level-order-traversal.cpp
#include "../std.hpp";
class Solution {
public:
    vector<vector<int>> ret;
    vector<vector<int>> levelOrder(TreeNode *root)
    {
        buildVec(root, 0);
        return ret;
    }
    void buildVec(TreeNode *root, int depth)
    {
        if (!root) {
            return;
        }
        if (ret.size() == depth) {
            ret.resize(depth + 1);
        }
        ret[depth].push_back(root->val);
        buildVec(root->left, depth + 1);
        buildVec(root->right, depth + 1);
    }
};

File: js/Trees/102.binary-tree-level-order-traversal.ts
function levelOrder(root: TreeNode | null): number[][] {
  let res: number[][] = []
  function dfs(root: TreeNode | null, depth: number) {
    if (!root) {
      return
    }
    if (res.length == depth) {
      res.push([])
    }
    res[depth].push(root.val)
    dfs(root.left, depth + 1)
    dfs(root.right, depth + 1)
  }
  dfs(root, 0)
  return res
}

